kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -188,65 +262,20 @@ public void removeDomainMetadata(String domain) {
    *
    * @return A list of {@link DomainMetadata} containing domain metadata to be committed in this
    *     transaction.
    */
-  public List<DomainMetadata> getDomainMetadatas() {
-    // If we have already processed the domain metadatas, then return the list.
-    if (domainMetadatas.isPresent()) {
-      return domainMetadatas.get();
-    }
-    generateClusteringDomainMetadataIfNeeded();
-    if (domainMetadatasAdded.isEmpty() && domainMetadatasRemoved.isEmpty()) {
-      // If no domain metadatas are added or removed, return an empty list. This is to avoid
-      // unnecessary loading of the domain metadatas from the snapshot (which is an expensive
-      // operation).
-      domainMetadatas = Optional.of(Collections.emptyList());
-      return Collections.emptyList();
-    }
-
-    // Add all domain metadatas added in the transaction
-    List<DomainMetadata> finalDomainMetadatas = new ArrayList<>(domainMetadatasAdded.values());
-
-    // Generate the tombstones for the removed domain metadatas
-    Map<String, DomainMetadata> snapshotDomainMetadataMap = readSnapshot.getDomainMetadataMap();
-    for (String domainName : domainMetadatasRemoved) {
-      // Note: we know domainName is not already in finalDomainMetadatas because we do not allow
-      // removing and adding a domain with the same identifier in a single txn!
-      if (snapshotDomainMetadataMap.containsKey(domainName)) {
-        DomainMetadata domainToRemove = snapshotDomainMetadataMap.get(domainName);
-        if (domainToRemove.isRemoved()) {
-          // If the domain is already removed we throw an error to avoid any inconsistencies or
-          // ambiguity. The snapshot read by the connector is inconsistent with the snapshot
-          // loaded here as the domain to remove no longer exists.
-          throw new DomainDoesNotExistException(
-              dataPath.toString(), domainName, readSnapshot.getVersion());
-        }
-        finalDomainMetadatas.add(domainToRemove.removed());
-      } else {
-        // We must throw an error if the domain does not exist. Otherwise, there could be unexpected
-        // behavior within conflict resolution. For example, consider the following
-        // 1. Table has no domains set in V0
-        // 2. txnA is started and wants to remove domain "foo"
-        // 3. txnB is started and adds domain "foo" and commits V1 before txnA
-        // 4. txnA needs to perform conflict resolution against the V1 commit from txnB
-        // Conflict resolution should fail but since the domain does not exist we cannot create
-        // a tombstone to mark it as removed and correctly perform conflict resolution.
-        throw new DomainDoesNotExistException(
-            dataPath.toString(), domainName, readSnapshot.getVersion());
-      }
-    }
-    domainMetadatas = Optional.of(finalDomainMetadatas);
-    return finalDomainMetadatas;
+  public List<DomainMetadata> getComputedDomainMetadatasToCommit() {
```

Comment:
nit: delete this method, and just use `domainMetadataState.getComputedDomainMetadatasToCommit()` directly -- and move this comment to that method? what do you think?